concent_api/concent_api/tests/test_middleware.py

```diff
@@ -0,0 +1,51 @@
+from base64                 import b64encode
+import os
+
+from django.test            import override_settings
+from django.test            import TestCase
+from django.urls            import reverse
+from golem_messages         import dump
+from golem_messages         import message
+
+from git import Repo
+
+from utils.testing_helpers  import generate_ecc_key_pair
+
+
+(CONCENT_PRIVATE_KEY,   CONCENT_PUBLIC_KEY)   = generate_ecc_key_pair()
+(PROVIDER_PRIVATE_KEY,  PROVIDER_PUBLIC_KEY)  = generate_ecc_key_pair()
+
+
+@override_settings(
+    CONCENT_PRIVATE_KEY    = CONCENT_PRIVATE_KEY,
+    CONCENT_PUBLIC_KEY     = CONCENT_PUBLIC_KEY,
+)
+class GolemMessagesVersionMiddlewareTest(TestCase):
+
+    def test_golem_messages_version_middleware_attach_http_header_to_response(self):
+        """
+        Tests that response from Concent:
+
+        * Contains HTTP header 'Concent-Golem-Messages-Version'.
+        * Header contains latest git commit id of golem_messages package.
+        """
+        ping_message = message.MessagePing()
+        serialized_ping_message = dump(ping_message, PROVIDER_PRIVATE_KEY, CONCENT_PUBLIC_KEY)
+
+        response = self.client.post(
+            reverse('core:send'),
+            data                           = serialized_ping_message,
+            content_type                   = 'application/octet-stream',
+            HTTP_CONCENT_CLIENT_PUBLIC_KEY = b64encode(PROVIDER_PUBLIC_KEY).decode('ascii'),
+        )
+
+        self.assertFalse(str(response.status_code).startswith('5'))
```

Comment:
What would you say about
``` python
self.assertTrue(500 <= response.status_code < 600)
```

or

```
self.assertGreaterEqual(response.status_code, 500)
self.assertLess(response.status_code, 600)
```
which is, admittedly, longer and a bit less readable but gives more meaningful error messages if a test fails.

Reply:
:+1:

Reply:
One more thing. Now that I think about it, this check will never fail. That's because in tests Django does not catch view exceptions to replace them with HTTP 500. If there's an exception in the view, it simply escapes and interrupts the test.

In general, we never test for 5xx errors (or their absence) in tests.

Reply:
It was just added to confirm that we get expected response from Concent.